core/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -122,8 +120,11 @@
     protected final ReleasableLock writeLock;
     private final Path location;
     private TranslogWriter current;
-    private volatile ImmutableTranslogReader currentCommittingTranslog;
-    private volatile long lastCommittedTranslogFileGeneration = -1; // -1 is safe as it will not cause an translog deletion.
+
+    private final static long NOT_SET_GENERATION = -1;
+
+    private volatile long currentCommittingGeneration = NOT_SET_GENERATION; // -1 is safe as it will not cause an translog deletion.
```

Comment:
Move this comment up to the `NOT_SET_GENERATION` constant?  And "an translog" -> "a translog".